lib/python/src/bailo/helper/release.py

```diff
@@ -132,23 +134,52 @@ def from_version(cls, client: Client, model_id: str, version: Version | str) ->
             draft,
         )
 
-    def download(self, filename: str) -> Any:
+    def download(self, filename: str, write: bool | None = True, path: str | None = None) -> Any:
         """Give returns a Reading object given the file id.
 
         :param filename: The name of the file to retrieve
+        :param write: Bool to determine if writing file to disk, defaults to True
+        :param path: Local path to write file to (if write set to True)
+
         :return: A JSON response object
         """
-        return self.client.get_download_by_filename(self.model_id, str(self.version), filename)
+        res = self.client.get_download_by_filename(self.model_id, str(self.version), filename)
+
+        if write:
+            if path is None:
+                path = filename
+            with open(path, "wb") as f:
+                f.write(res.content)
 
-    def upload(self, name: str, file: BytesIO) -> str:
-        """Upload files in a given directory to the release.
+        return res
 
-        :param name: The name of the file to upload to bailo
-        :param f: A BytesIO object
+    def upload(self, path: str, data: BytesIO | None = None) -> str:
```

Comment:
instead of downloading mlflow to the local directory, download to temporary directories instead.  (e.g. using tempfiles modules)